target/scripts/helper-functions.sh

```diff
@@ -252,8 +303,13 @@ function _obtain_hostname_and_domainname
   fi
 }
 
+# Call this method when you want to panic (emit a 'FATAL' log level error, and exit uncleanly).
+# `dms_panic` methods should be preferred if your failure type is supported.
 function _shutdown
 {
+  local FATAL_ERROR_MESSAGE=$1
+
+  _notify 'fatal' "${FATAL_ERROR_MESSAGE}"
   _notify 'err' "Shutting down.."
   kill 1
```

Comment:
This is fine with me, but we could use

``` suggestion
  kill -TERM 1
```

instead to make clear we're sending `SIGTERM` here. Otherwise, I'd have to always look it up (which is okay though. Just a small hint)

Reply:
To keep it uniform with other `kill` calls in DMS, `-SIGTERM` should be used instead of `-TERM`.

On the other hand, why providing a functionless option (because it's default), which should be well known by everyone, that ever used kill? This will add 5-8 extra bytes to our image!!°1^1^1 😄

Reply:
I'll add this in a small follow up PR or if @casperklein takes on the other improvement replacing `_defunc` I guess it could go with that.